addons/mail/models/mail_activity_plan_template.py

```diff
@@ -71,6 +97,13 @@ def _compute_responsible_id(self):
             if template.responsible_type != 'other' and template.responsible_id:
                 template.responsible_id = False
 
+    def _calculate_date_deadline(self, plan_date=False):
+        """ Return the deadline of the activity to be created from the plan date. """
+        self.ensure_one()
+        base = plan_date or fields.Date.context_today(self)
+        sign = 1 if self.delay_from == 'after_schedule_date' else -1
```

Comment:
Can't you instead first compute the delta, and then inside a if/else add or subtract it from the base_date?

Would be the same but probably easier to read.